app/services.go

```diff
@@ -0,0 +1,153 @@
+package app
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"sort"
+
+	"github.com/karimra/gnoic/api"
+	"github.com/olekukonko/tablewriter"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"google.golang.org/grpc/metadata"
+	reflectpb "google.golang.org/grpc/reflection/grpc_reflection_v1alpha"
+)
+
+type reflectionResponse struct {
+	TargetError
+	rsp *reflectpb.ServerReflectionResponse
+}
+
+func (a *App) InitServicesFlags(cmd *cobra.Command) {
+	cmd.ResetFlags()
+	//
+	cmd.Flags().VisitAll(func(flag *pflag.Flag) {
+		a.Config.FileConfig.BindPFlag(fmt.Sprintf("%s-%s", cmd.Name(), flag.Name), flag)
+	})
+}
+
+func (a *App) RunEServices(cmd *cobra.Command, args []string) error {
+	targets, err := a.GetTargets()
+	if err != nil {
+		return err
+	}
+	numTargets := len(targets)
+	responseChan := make(chan *reflectionResponse, numTargets)
+	a.wg.Add(numTargets)
+	for _, t := range targets {
+		go func(t *api.Target) {
+			defer a.wg.Done()
+			ctx, cancel := context.WithCancel(a.ctx)
+			defer cancel()
+			ctx = metadata.AppendToOutgoingContext(ctx, "username", *t.Config.Username, "password", *t.Config.Password)
+
+			err = t.CreateGrpcClient(ctx, a.createBaseDialOpts()...)
+			if err != nil {
+				responseChan <- &reflectionResponse{
+					TargetError: TargetError{
+						TargetName: t.Config.Name,
+						Err:        err,
+					},
+				}
+				return
+			}
+			defer t.Close()
+
+			rfc := reflectpb.NewServerReflectionClient(t.Conn())
+			info, err := rfc.ServerReflectionInfo(ctx)
+			if err != nil {
+				responseChan <- &reflectionResponse{
+					TargetError: TargetError{
+						TargetName: t.Config.Name,
+						Err:        err,
+					},
+				}
+				return
+			}
+			err = info.Send(&reflectpb.ServerReflectionRequest{
+				MessageRequest: &reflectpb.ServerReflectionRequest_ListServices{},
+			})
+			if err != nil {
+				responseChan <- &reflectionResponse{
+					TargetError: TargetError{
+						TargetName: t.Config.Name,
+						Err:        err,
+					},
+				}
+				return
+			}
+			rsp, err := info.Recv()
+
+			responseChan <- &reflectionResponse{
+				TargetError: TargetError{
+					TargetName: t.Config.Name,
+					Err:        err,
+				},
+				rsp: rsp,
+			}
+		}(t)
+	}
+	a.wg.Wait()
+	close(responseChan)
+	errs := make([]error, 0, numTargets)
+	result := make([]*reflectionResponse, 0, numTargets)
+
+	for rsp := range responseChan {
+		if rsp.Err != nil {
+			wErr := fmt.Errorf("%q Services failed: %v", rsp.TargetName, rsp.Err)
+			a.Logger.Error(wErr)
+			errs = append(errs, wErr)
+			continue
+		}
+		result = append(result, rsp)
+		a.printMsg(rsp.TargetName, rsp.rsp)
+	}
+
+	switch a.Config.Format {
```

Comment:
also i would probably pull this into it's own function as then it would be easier to test